compiler/visq/visqlib/QErrorComputer.py

```diff
@@ -104,41 +120,53 @@ def run(self):
                 # To prevent this, relaxed PEIR with epsilon(10^(-6)) is used.
                 rPEIR = PEAK_ERROR / (INTERVAL + 0.000001)
 
-                if tensor_name in qerror_map:
-                    qerror_map[tensor_name] += rPEIR
+                if tensor_name in self.qerror_map:
+                    self.qerror_map[tensor_name] += rPEIR
                 else:
-                    qerror_map[tensor_name] = rPEIR
+                    self.qerror_map[tensor_name] = rPEIR
 
-        for tensor_name, acc in qerror_map.items():
-            qerror_map[tensor_name] = acc / self._num_data
+    def get_final_result(self):
+        qerror_map = dict()
+        for tensor_name, acc in self.qerror_map.items():
+            qerror_map[tensor_name] = acc / self._num_processed_data
 
         return qerror_map
 
+    def run(self, fp32_dir=None, fq_dir=None):
+        self.advance_on(fp32_dir, fq_dir)
+        return self.get_final_result()
+
 
 class MSEComputer(QErrorComputer):
     def __init__(self, fp32_dir, fq_dir):
         super().__init__(fp32_dir, fq_dir)
+        self.qerror_min = float('inf')
+        self.qerror_max = -self.qerror_min
 
-    def run(self):
-        qerror_map = dict()
-        qerror_min = float('inf')
-        qerror_max = -qerror_min
+    def advance_on(self, fp32_dir, fq_dir):
+        self.update_dirs(fp32_dir, fq_dir)
         for tensor_name, data_path in self._data_path.items():
             for (fp32_data_path, fq_data_path) in data_path:
                 fp32_data = np.load(fp32_data_path)
                 fq_data = np.load(fq_data_path)
 
                 MSE = np.square(fp32_data - fq_data).mean()
 
-                if tensor_name in qerror_map:
-                    qerror_map[tensor_name] += MSE
+                if tensor_name in self.qerror_map:
+                    self.qerror_map[tensor_name] += MSE
                 else:
-                    qerror_map[tensor_name] = MSE
+                    self.qerror_map[tensor_name] = MSE
 
-                qerror_min = min(MSE, qerror_min)
-                qerror_max = max(MSE, qerror_max)
+                self.qerror_min = min(MSE, self.qerror_min)
+                self.qerror_max = max(MSE, self.qerror_max)
+
+    def get_final_result(self):
+        qerror_map = dict()
+        for tensor_name, acc in self.qerror_map.items():
+            qerror_map[tensor_name] = acc / self._num_processed_data
 
-        for tensor_name, acc in qerror_map.items():
-            qerror_map[tensor_name] = acc / self._num_data
+        return qerror_map, self.qerror_min, self.qerror_max
 
-        return qerror_map, qerror_min, qerror_max
+    def run(self, fp32_dir=None, fq_dir=None):
```

Comment:
Why `run` has fp32_dir and fq_dir ? It seems that they are not used.

Reply:
Ahhh.. sorry. I'll remove them.

Reply:
fixed.